src/core/contracts/identity.ts

```diff
@@ -323,3 +326,194 @@ export const upsertDelegate = async (
 
   return contract.connect(signer).delegate(address, role);
 };
+
+/**
+ * removeDelegate() Add or change permissions for delegate
+ *
+ * @param contractAddress - Address of the identity contract to use
+ * @param address - Address of delegate to add permissions to
+ * @param endBlock - The block number to remove delegate
+ * @param opts - Optional. Configuration overrides, such as from address, if any
+ */
+export const removeDelegate = async (
+  contractAddress: EthereumAddress,
+  address: EthereumAddress,
+  endBlock: DelegationRole,
+  opts?: ConfigOpts
+): Promise<ContractTransaction> => {
+  const signer = requireGetSigner(opts);
+  const provider = requireGetProvider(opts);
+  const contract = await Identity__factory.connect(contractAddress, provider);
+
+  return contract.connect(signer).delegateRemove(address, endBlock);
+};
+
+const IDENTITY_DECODER = new ethers.utils.Interface(Identity__factory.abi);
+
+export const siftDelegateLogs = (logs: ParsedLog[]): DelegateLogData[] => {
+  return logs.map((item: ParsedLog) => {
+    const {
+      fragment: {
+        args: { delegate, endBlock },
+        name,
+      },
+      log: { address, blockNumber },
+    } = item;
+
+    const eventData: DelegateLogData = {
+      name,
+      identityAddress: address,
+      delegate: delegate,
+      blockNumber,
+    };
+
+    if (name === "DSNPRemoveDelegate") {
+      eventData.endBlock = endBlock.toNumber();
+    }
+
+    return eventData;
+  });
+};
+
+const groupByIdentityAddress = (delegateLogs: DelegateLogData[]): Record<string, DelegateLogData[]> => {
+  return delegateLogs.reduce((acc: unknown | any, current: DelegateLogData) => {
+    if (!acc[current.identityAddress]) {
+      acc[current.identityAddress] = [];
+    }
+
+    acc[current.identityAddress].push(current);
+
+    return acc;
+  }, {});
+};
+
+/**
+ * dsnpAddDelegateFilter() Retrieves event filter for DSNPAddDelegate event
+ *
+ * @param publicAddress - Address to filter on
+ * @param opts - Optional. Configuration overrides, such as from address, if any
+ * @returns DSNPBatch event filter
+ */
+export const getDelegateIdentitiesFrom = async (
```

Comment:
maybe this is a better name: findIdentitiesByPublicAddress, getIdentitiesByPublicAddress, findDelegateIdentitiesFrom,

Reply:
merged